test/tests-using.js

```diff
@@ -0,0 +1,2485 @@
+        "end": 20,
+        "id": {
+          "type": "Identifier",
+          "start": 8,
+          "end": 9,
+          "name": "x"
+        },
+        "init": {
+          "type": "Identifier",
+          "start": 12,
+          "end": 20,
+          "name": "resource"
+        }
+      }],
+      "kind": "using"
+    }]
+  }],
+  "sourceType": "script"
+}, {"ecmaVersion": 17});
+
+// Multiple variable declarations
+test("{ using a = resource1, b = resource2; }", {
+  "type": "Program",
+  "start": 0,
+  "end": 39,
+  "body": [{
+    "type": "BlockStatement",
+    "start": 0,
+    "end": 39,
+    "body": [{
+      "type": "VariableDeclaration",
+      "start": 2,
+      "end": 37,
+      "declarations": [{
+        "type": "VariableDeclarator",
+        "start": 8,
+        "end": 21,
+        "id": {
+          "type": "Identifier",
+          "start": 8,
+          "end": 9,
+          "name": "a"
+        },
+        "init": {
+          "type": "Identifier",
+          "start": 12,
+          "end": 21,
+          "name": "resource1"
+        }
+      }, {
+        "type": "VariableDeclarator",
+        "start": 23,
+        "end": 36,
+        "id": {
+          "type": "Identifier",
+          "start": 23,
+          "end": 24,
+          "name": "b"
+        },
+        "init": {
+          "type": "Identifier",
+          "start": 27,
+          "end": 36,
+          "name": "resource2"
+        }
+      }],
+      "kind": "using"
+    }]
+  }],
+  "sourceType": "script"
+}, {"ecmaVersion": 17});
+
+// Using with array syntax - this should be parsed as member expression assignment using[a, b] = arr
+test("{ using [a, b] = arr; }", {
+  type: "Program",
+  start: 0,
+  end: 23,
+  body: [{
+    type: "BlockStatement",
+    start: 0,
+    end: 23,
+    body: [{
+      type: "ExpressionStatement",
+      start: 2,
+      end: 21,
+      expression: {
+        type: "AssignmentExpression",
+        start: 2,
+        end: 20,
+        operator: "=",
+        left: {
+          type: "MemberExpression",
+          start: 2,
+          end: 14,
+          object: {
+            type: "Identifier",
+            start: 2,
+            end: 7,
+            name: "using"
+          },
+          computed: true,
+          property: {
+            type: "SequenceExpression",
+            start: 9,
+            end: 13,
+            expressions: [
+              {
+                type: "Identifier",
+                start: 9,
+                end: 10,
+                name: "a"
+              },
+              {
+                type: "Identifier",
+                start: 12,
+                end: 13,
+                name: "b"
+              }
+            ]
+          }
+        },
+        right: {
+          type: "Identifier",
+          start: 17,
+          end: 20,
+          name: "arr"
+        }
+      }
+    }]
+  }],
+  sourceType: "script"
+}, {"ecmaVersion": 17});
+
+// Top-level using declaration
+test("using x = resource;", {
```

Comment:
From what I understand, this is an early error, because I don't think we can have a top-level `using` declaration for a script. However, I think it is allowed for modules.

<img width="1232" alt="image" src="https://github.com/user-attachments/assets/77547427-acfb-4c11-91ce-8d932bd07ad2" />

Reply:
fixed: https://github.com/acornjs/acorn/pull/1369/commits/5f1f1246de7e9165401ef0c18d105b23a75fca0e